pyface/data_view/data_view_errors.py

```diff
@@ -0,0 +1,27 @@
+# (C) Copyright 2005-2020 Enthought, Inc., Austin, TX
+# All rights reserved.
+#
+# This software is provided without warranty under the terms of the BSD
+# license included in LICENSE.txt and may be redistributed only under
+# the conditions described in the aforementioned license. The license
+# is also available online at http://www.enthought.com/licenses/BSD.txt
+#
+# Thanks for using Enthought open source!
+
+""" Provide an Exception classes for the the DataView code.
+"""
+
+
+class DataViewError(ValueError):
```

Comment:
Why does this subclass `ValueError`?  This means if we do `except ValueError`, all errors related to DataView will be captured. This is a design choice that would be difficult to revert.  Is there a particular reason why we should categorize all data view errors as a type of ValueError? If not, it would be more flexible to defer this by not subclassing `ValueError`. e.g. `TraitError` just subclasses `Exception`.

Reply:
No strong reason: `ValueError` is described as an error type to return when you have an inappropriate value, but no more specific exception applies (eg. `IndexError`) which fits this situation, and `ValueError` is not a leaf of the `Exception` hierarchy.  In particular, the `DataViewSetError` is often a proxy for a different `ValueError` generated by eg. a failure to evaluate text to an appropriate data value, or setting a data value of the wrong type.

So that's why the original choice was made.

By having a base class for all the `DataViewErrors`, it makes it easy to filter out from other `ValueError`s if someone really cares about it.

And if it is wrong, I don't see why it is a particularly difficult to change.

Let's keep it as-is for now.

Reply:
I don't think I follow that, e.g. why `DataViewSetError` can't be defined as `DataViewSetError(DataViewError, ValueError)` whereas `DataViewError` is simply `DataViewError(Exception)`. This will allow the flexibility to have specialized exception that is not related to `ValueError`, e.g. `DataViewGetError(DataViewError, LookupError)`.  Up to you if you don't want this. 

>  I don't see why it is a particularly difficult to change.

I guess I am thinking about these exceptions being used in try-except in code outside of pyface. Then it would be backward incompatible to change it one way or another.